### DISCUSSION TITLE
Learning To Code

### DISCUSSION THREAD

**Discussion by EvDenny** [not-a-ethan/My-work#8] [Q&A] [answered]
Hey man! Saw your website and reminded me of one of my own private repositories that I'm working on to demonstrate my ability in C++, C, JS, HTML, CSS, Python, and maybe Ruby. I'm trying to diversify my programming skills by learning a bunch of languages so I understand how to work on bigger projects that are built in different ways. I just wanted to ask if you could write up a summary for me of how you put your website together, what your workflow was like, how you overcame challenges, and if you wanted to collaborate on some project sometime. I really want to start working with other programmers to learn Git and get ready for making a career out of this. **let me know if you want to see some of my work recently**. I'm a little disorganized right now and I'm working on it. Got codespaces for free for another 5 days to take advantage of. Anyway. Lmk what you think. I think your website is dope btw.

**Comment by not-a-ethan** [comment, marked as answer]
## For personal projects
For most changes, I just directly commit to the [`main`](https://github.com/ethan-master-coding/My-work/tree/main) branch once I tested the changes locally. For large changes, I open a PR and review the code a day or more later to make sure I did not make any stupid changes before I merge it into [`main`](https://github.com/ethan-master-coding/My-work/tree/main).

## For other projects

For other projects like [Coin Catalog](https://github.com/Coin-Catalog) I follow the rules I set like always making a PR. I will never commit directly to `main` unless it's a new repo I am setting up or if I am trying to fix a broken workflow.

**Reply by EvDenny** [reply]
I just took a look at your Coin Catalog site and I think I could do a little work to it to make it more responsive. Especially with images. I used to run the website for WSHU. You can actually find my work on the site if it's still there. Just search "Evan Denny".

Usually images for webpages need to be certain formats and sizes to load properly without caching the sitemap. Since that's done automatically for most people people tend to forget to resize and reformat their images. You don't need all that detail when the screen isn't as large as a DSLR image lol. 

I'll make a pull request at some point and see what I can do. Are their any projects you're working on that you need help with? 

I could use some help with a C++ program I'm writing that has a really strange bug. I don't know if you're familiar with C++ but if you do I could use a second pair of eyes. Or if you know anyone. 

At the same time I'm going to start a website repository soon for my projects and I'd like some help if you would be willing. Thanks for the advice. I was just commuting everything to main. But I make incremental changes so it hasn't been a huge issue yet. 

My plan is to consolidate all my C++ programs into a single executable to download and so far I've been doing well. Two working games and a placeholder program to setup the menu system and working on a bunch of things. I'm a little ADHD so I tend to work on a lot of projects at once. I write them down on paper in pseudo-code first. It really helps me when it comes to actually building the project when you have it all planned out.
